packages/hoppscotch-common/src/composables/whats-new.ts

```diff
@@ -0,0 +1,41 @@
+import { toast as sonner } from "@hoppscotch/ui"
+import { markRaw } from "vue"
+import WhatsNewDialog from "~/components/app/WhatsNewDialog.vue"
+import { getService } from "~/modules/dioc"
+import { PersistenceService } from "~/services/persistence"
+import { version } from "./../../package.json"
+
+export const useWhatsNewDialog = async function () {
+  const persistenceService = getService(PersistenceService)
+
+  const oldVersion = persistenceService.getLocalConfig("hopp_v")
+  const latestMajorVersion = version.split(".").slice(0, 2).join(".")
+
+  if (oldVersion === version) return
+
+  // Skipping the minor version update. e.g. 2024.1.0 -> 2024.1.1
+  // Checking major version update. e.g. 2024.1 -> 2024.2
+
+  const isUpdatedToLatestMajorVersion =
+    oldVersion?.split(".").slice(0, 2).join(".") === latestMajorVersion
+
+  if (isUpdatedToLatestMajorVersion) {
```

Comment:
Is this check necessary? Doesn't this resolve to the same check (`oldVersion` & `version`) above where the early return happens?

`latestMajorVersion` -> `version.split(".").slice(0, 2).join(".")`

Reply:
Got the use case. But, during a minor bump, won't the above check result in `true` while comparing against the major version strings alone? Say, while updating from `2024.3.3` -> `2024.3.4`. If there's already an entry under `localStorage` at first while loading the app, it hits the early return since both the versions (`localStorage` entry & `@hoppscotch/common` version) are the same. An update was detected via the `registerSw` flow. Now, in the above flow, the versions are not the same since the `@hoppscotch/common` version is bumped. Obtaining the major version entries now points to the same value, the `isUpdatedToLatestMajorVersion` check would resolve to `true` entering the block about triggering the network call to fetch the release notes.

Now for a major bump, `2024.3.0` -> `2024.6.0`, the same flow happens and post the app reload via `registerSw`, the `isUpdatedToLatestMajorVersion` check won't hold `true` since the major versions (`2024.3` & `2024.6`) are different. This also applies when the `localStorage` entry isn't present and the update happens.

Also, could you elaborate on how the `whats-new` module will be triggered after performing an update? Being a module it runs when the app is loaded at first, but we'll require it to be triggered after an update happens via the `registerSw` flow as well.

https://github.com/hoppscotch/hoppscotch/blob/6e63c723b7218f3d8db91f01401bbc116fa46dee/packages/hoppscotch-common/src/modules/pwa.ts#L20-L22

Reply:
Ahh! I see. It's fixed, now!